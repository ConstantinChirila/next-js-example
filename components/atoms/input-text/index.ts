export * from './input-text'
export * from './input-text.styled'
export * from './input-text.types'
