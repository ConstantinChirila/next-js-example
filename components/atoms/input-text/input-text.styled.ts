import styled from 'styled-components'

import { StyledSVGIcon, TInputTextProperties } from '@tooploox-test/components'

export const StyledInputTextWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: flex-start;
  position: relative;
  width: 100%;
`

export const StyledLabel = styled.label`
  color: ${({ theme }) => theme.colors.grey4};
  font-size: 1.4rem;
  left: 1rem;
  letter-spacing: 0.04rem;
  pointer-events: none;
  position: absolute;
  top: 1rem;
  transition: 0.1s ease transform, color, font-size;
`

export const StyledIconWrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.grey1};
  border-radius: ${({ theme }) => theme.radius.md};
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
  display: flex;
  height: 3.6rem;
  justify-content: center;
  padding-left: 0.5rem;
  pointer-events: none;
  width: 2.5rem;

  > ${StyledSVGIcon} {
    transform: scale(0.7);
  }

  ~ ${StyledLabel} {
    left: 3rem;
  }
`

export const StyledInputText = styled.input<TInputTextProperties>`
  background-color: ${({ theme }) => theme.colors.grey1};
  border-radius: ${({ theme }) => theme.radius.md};
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border: none;
  color: ${({ theme }) => theme.colors.grey5};
  font-family: ${({ theme }) => theme.font.family.primary}, sans-serif;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 1.4rem 1.4rem 1.4rem 0.8rem;
  width: 100%;

  ::placeholder {
    /* Why have done it?
     This is due to the design of the form element and a lack of time to find any better solution :-(
     Typically, I would just keep it as it has some benefits to give a hint for people what they need to enter in the field.
     I needed to use this trick... Sorry!
     */
    color: transparent;
  }

  &:focus,
  &:not(:placeholder-shown) {
    + ${StyledLabel} {
      color: ${({ theme }) => theme.colors.navy};
      font-size: 1rem;
      transform: translate3d(0, -1rem, 0);
    }
  }
`
