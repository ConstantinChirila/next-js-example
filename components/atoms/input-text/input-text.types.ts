import { HTMLAttributes, ReactElement } from 'react'

export type TInputTextIconTypes = {
  icon?: ReactElement
}

export type TInputTextRequiredTypes = {
  id: string
  placeholder?: string
}

export type TInputTextProperties = HTMLAttributes<HTMLInputElement> &
  TInputTextIconTypes &
  TInputTextRequiredTypes
