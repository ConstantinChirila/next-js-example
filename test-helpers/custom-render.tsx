import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
