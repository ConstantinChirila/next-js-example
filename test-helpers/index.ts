export * from './custom-render'
export * from './setup'
