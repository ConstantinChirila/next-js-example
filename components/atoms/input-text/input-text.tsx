import React, { ReactElement } from 'react'

import {
  StyledIconWrapper,
  StyledInputText,
  StyledInputTextWrapper,
  StyledLabel,
} from './input-text.styled'
import { TInputTextProperties } from './input-text.types'

export function InputText({
  id,
  placeholder,
  icon,
  ...properties
}: TInputTextProperties): ReactElement {
  return (
    <StyledInputTextWrapper>
      {icon && <StyledIconWrapper>{icon}</StyledIconWrapper>}
      <StyledInputText {...properties} id={id} placeholder={placeholder} />
      <StyledLabel htmlFor={id}>{placeholder}</StyledLabel>
    </StyledInputTextWrapper>
  )
}
