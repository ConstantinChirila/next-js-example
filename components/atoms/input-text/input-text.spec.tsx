import React from 'react'

import { InputText } from './input-text'
import { renderWithTheme } from '../../../test-helpers'

describe('InputText', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(
      <InputText placeholder="yoyo" id="yoyo" />
    )
    expect(container).toMatchSnapshot()
  })
})
