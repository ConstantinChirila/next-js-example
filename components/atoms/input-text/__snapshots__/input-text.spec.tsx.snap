// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should match a snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  width: 100%;
}

.c3 {
  color: #9E9E9E;
  font-size: 1.4rem;
  left: 1rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  pointer-events: none;
  position: absolute;
  top: 1rem;
  -webkit-transition: 0.1s ease -webkit-transform,color,font-size;
  -webkit-transition: 0.1s ease transform,color,font-size;
  transition: 0.1s ease transform,color,font-size;
}

.c1 {
  background-color: #F0F3F4;
  border-radius: 0.8rem;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border: none;
  color: #4F4F4F;
  font-family: Roboto,sans-serif;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 1.4rem 1.4rem 1.4rem 0.8rem;
  width: 100%;
}

.c1::-webkit-input-placeholder {
  color: transparent;
}

.c1::-moz-placeholder {
  color: transparent;
}

.c1:-ms-input-placeholder {
  color: transparent;
}

.c1::placeholder {
  color: transparent;
}

.c1:focus + .c2,
.c1:not(:placeholder-shown) + .c2 {
  color: #452CDC;
  font-size: 1rem;
  -webkit-transform: translate3d(0,-1rem,0);
  -ms-transform: translate3d(0,-1rem,0);
  transform: translate3d(0,-1rem,0);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      id="yoyo"
      placeholder="yoyo"
    />
    <label
      class="c2 c3"
      for="yoyo"
    >
      yoyo
    </label>
  </div>
</div>
`;
